### PR TITLE
RAB-613: Reorganize Channel's back-end code to comply with ADR

### PR DESCRIPTION
**Pull request by grena**
**Description (for Contributor and Core Developer)**

What is done in this PR : 
- Create a new BC in src/Akeneo/Channel
- Remove the Locale subdomain and merge it into the Channel BC
- Add dummy tests for Integration and Acceptance
- Make the tests working in the Channel BC (I had to add exclude rules for linting legacy classes)
- Make the tests working outside the Channel BC
- Add the tests targets to the test.mk file

This PR does not contains a lot of refactoring :
For now, we just spread out `Bundle` subdirs into `Infrastructure` and move the `Component` dir into `Infrastructure`.
Some PRs will be done soon to refactor and decouple this codebase.

cf https://www.notion.so/akeneo/6796d7ffa19047e88c0fac34dc039786?v=513930ac3763424f9affc9ba6bd67c67&p=54d8fbe052de4709999bebca5e200f12

**Definition Of Done (for Core Developer only)**

- [x] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -21,7 +21,9 @@
             "Akeneo\\Platform\\Job\\": "src/Akeneo/Platform/Job/back",
             "Akeneo\\FreeTrial\\": "src/Akeneo/FreeTrial/back",
             "Akeneo\\Pim\\Enrichment\\Product\\": "src/Akeneo/Pim/Enrichment/Product/back",
-            "Akeneo\\Pim\\Enrichment\\Category\\": "src/Akeneo/Pim/Enrichment/Category/back"
+            "Akeneo\\Pim\\Enrichment\\Category\\": "src/Akeneo/Pim/Enrichment/Category/back",
+            "Akeneo\\Channel\\": "src/Akeneo/Channel/back",
+            "Akeneo\\Channel\\Component\\": "src/Akeneo/Channel/Component"
```
Comment (grena) [inline composer.json:26]: TODO: to revert, I did this to have the PIM working while moving Infrastructure files.

File: src/Akeneo/Channel/back/Infrastructure/Controller/ExternalApi/ChannelController.php [hunk 2]
```diff
@@ -35,90 +35,31 @@
...
-     * @throws NotFoundHttpException
-     *
-     * @return JsonResponse
-     *
-     * @AclAncestor("pim_api_channel_list")
-     */
-    public function getAction(Request $request, $code)
+    public function getAction(Request $request, string $code): JsonResponse
     {
+        if (!$this->securityFacade->isGranted('pim_api_channel_list')) {
```
Comment (ghoussard) [inline src/Akeneo/Channel/back/Infrastructure/Controller/ExternalApi/ChannelController.php:56]: As `AclAncestor` annotation didn't works anymore (I think the load of annotations is broken if the `Controller` dir is not under a `Bundle` dir). I injected the `SecurityFacade` into controllers that needed it.
I took the opportunity to clean a bit thoses controllers : remove unused PHPDoc, add typehint and use the PHP8 construct syntax.

File: src/Akeneo/Channel/back/tests/.php_cd.php [hunk 3]
```diff
@@ -5,26 +5,31 @@
...
-
+$finder->notPath('tests');
 $builder = new RuleBuilder();
 
 $rules = [
     $builder->only([
+        'Akeneo\Platform\Bundle\FrameworkBundle\Security\SecurityFacadeInterface',
+        'Oro\Bundle\SecurityBundle\Exception\AccessDeniedException',
+
+        // Rules from Akeneo\Channel\Bundle
```
Comment (ghoussard) [inline src/Akeneo/Channel/back/tests/.php_cd.php:16]: Here I keep the former rules from the legacy. I add rules at the top to pass the CI.
These rules will be cleaned when legacy classes will be cleaned.

**Review by ghoussard — APPROVED** [review]

File: make-file/channel.mk [hunk 4]
```diff
@@ -1,57 +1,38 @@
...
+.PHONY: channel-acceptance-back
+channel-acceptance-back: #Doc: launch PHPUnit acceptance tests for channel bounded context
+ifeq ($(CI),true)
+	APP_ENV=test_fake $(PHP_RUN) vendor/bin/phpunit -c src/Akeneo/Channel/back/tests --log-junit var/tests/phpunit/phpunit_$$(uuidgen).xml --testsuite Channel_Acceptance_Test
+else
+	APP_ENV=test_fake $(PHP_RUN) vendor/bin/phpunit -c src/Akeneo/Channel/back/tests --testsuite Channel_Acceptance_Test $(O)
+endif
+
+.PHONY: channel-ci-back
+channel-ci-back: channel-lint-back channel-coupling-back channel-unit-back channel-acceptance-back channel-integration-back
```
Comment (ahocquard) [inline make-file/channel.mk:35]: I don't know if it's on purpose, but it allows to do `make -j --output-sync channel-ci-back` to parallelize everything.

**Review by ahocquard — COMMENTED** [review]
Seems good to me!
Is it plan to move the tests in the future?



Nice job, moving component and bundle inside infra will make it cleaner.

**Review by StevenVAIDIE — APPROVED** [review]

File: src/Akeneo/Channel/back/Infrastructure/Component/Model/Channel.php [hunk 5]
```diff
@@ -1,8 +1,8 @@
 <?php
 
-namespace Akeneo\Channel\Component\Model;
+namespace Akeneo\Channel\Infrastructure\Component\Model;
```
Comment (docteurklein) [inline src/Akeneo/Channel/back/Infrastructure/Component/Model/Channel.php:3]: I don't get why models are in Infrastructure?

Comment (ghoussard) [reply src/Akeneo/Channel/back/Infrastructure/Component/Model/Channel.php:3]: Indeed there is no sense. We plan to refactor these classes in the future. For instance, these models will be replaced by models in our API.

File: make-file/channel.mk (hunk 4 above)
Comment (ghoussard) [reply make-file/channel.mk:27]: Interesting command, thanks! 👌🏻

**Comment by ghoussard** [comment]
> Seems good to me! Is it plan to move the tests in the future?
> 
> Nice job, moving component and bundle inside infra will make it cleaner.

Yes that's the plan 👌🏻

**Pull request merged**